src/main/kotlin/org/neo4j/graphql/BuildingEnv.kt

```diff
@@ -92,20 +92,66 @@ class BuildingEnv(val types: MutableMap<String, GraphQLType>) {
                     else -> addFilterType(getInnerFieldsContainer(typeDefinition), createdTypes)
                 }
 
-                Operators.forType(types[filterType] ?: typeDefinition).forEach { op ->
-                    val wrappedType: GraphQLInputType = when {
-                        op.list -> GraphQLList(GraphQLTypeReference(filterType))
-                        else -> GraphQLTypeReference(filterType)
-                    }
-                    builder.field(GraphQLInputObjectField.newInputObjectField()
-                        .name(op.fieldName(field.name))
-                        .type(wrappedType))
+                if (field.isRelationship()) {
+                    createRelationFilterFields(type, field, filterType, builder)
+                } else {
+                    FieldOperator.forType(types[filterType] ?: typeDefinition)
+                        .forEach { op -> addFilterField(builder, field, op, filterType) }
                 }
+
             }
         types[filterName] = builder.build()
         return filterName
     }
 
+    private fun createRelationFilterFields(type: GraphQLFieldsContainer, field: GraphQLFieldDefinition, filterType: String, builder: GraphQLInputObjectType.Builder) {
+        val list = field.type.isList()
+        addFilterField(builder, field, RelationOperator.EQ_OR_NOT_EXISTS, filterType,
```

Comment:
I think more of this functionality, incl the descriptions and the add field to build could be moved into RelOperator

Reply:
So it's all in the same place and not spread throughout the code

Reply:
??